File: RAI/Clase 7/java/Animal.java
//package RAI.Clase 7.java;

import java.time.LocalDate;

public class Animal {
    //Atributos
    private String especie;
    private String sexo;
    private String nombre;
    private LocalDate fechaNacimiento;

    //Métodos
    public Animal(String especie, String sexo, String nombre, LocalDate fechaNacimiento){
        this.especie = especie;
        this.sexo = sexo;
        this.nombre = nombre;
        this.fechaNacimiento = fechaNacimiento;
    }

    public Animal(String nombre){
        this.nombre = nombre;
    }

    public String getEspecie(){
        return this.especie;
    }

    public String getSexo(){
        return this.sexo;
    }

    public String getNombre(){
        return this.nombre;
    }

    public LocalDate getFechaNacimiento(){
        return this.fechaNacimiento;
    }

    public void setEspecie(String newEspecie){
        this.especie = newEspecie;
    }

    public void setSexo(String newSexo){
        this.sexo = newSexo;
    }

    public void setNombre(String newNombre){
        this.nombre = newNombre;
    }

    public void setFechaNacimiento(LocalDate newFechaNacimiento){
        this.fechaNacimiento = newFechaNacimiento;
    }

    //comportamiento
    public void come(String cosa){
        System.out.println(this.nombre + " está comiendo " + cosa);
    }

    public void come(){
        System.out.println(this.nombre + " está comiendo algo");
    }

    public void haceSonido(){
        System.out.println(this.nombre + " hace un sonido...");
    }
}

File: RAI/Clase 7/python/Animal.py
class Animal:
    def __init__(self, especie, sexo, nombre, fecha_nacimiento=None):
        self.especie = especie
        self.sexo = sexo
        self.nombre = nombre
        self.fecha_namiciento = fecha_nacimiento

    def get_especie(self):
        return self.especie

    def get_sexo(self):
        return self.sexo

    def get_nombre(self):
        return self.nombre
    
    def get_fecha_nacimiento(self):
        return self.fecha_namiciento

    def set_especie(self, new_especie):
        self.especie = new_especie

    def set_sexo(self, new_sexo):
        self.sexo = new_sexo

    def set_nombre(self, new_nombre):
        self.nombre = new_nombre

    def set_fecha_nacimiento(self, new_fecha_nacimiento):
        self.fecha_nacimiento = new_fecha_nacimiento

    # comportamiento

    def come(self, cosa=None):
        if cosa == None:
            print(f"{self.nombre} come algo")
        else:
            print(f"{self.nombre} come {cosa}")

    def hace_sonido(self):
        print(f"{self.nombre} hace un sonido...")